compiler/packages/babel-plugin-react-compiler/src/ReactiveScopes/ExtractScopeDeclarationsFromDestructuring.ts

```diff
@@ -150,15 +152,7 @@ function transformDestructuring(
     if (!reassigned.has(place.identifier.id)) {
       return place;
     }
-    const tempId = state.env.nextIdentifierId;
-    const temporary = {
-      ...place,
-      identifier: {
-        ...place.identifier,
-        id: tempId,
```

Comment:
This was reusing the declarationId, but we don't strictly need to copy anything from `place` except the location. The main other thing we might want to preserve is the type, but we don't really use types after this in the pipeline.

Reply:
done!